Userland/Libraries/LibJS/Runtime/DatePrototype.cpp

```diff
@@ -144,6 +148,48 @@ JS_DEFINE_NATIVE_FUNCTION(DatePrototype::get_hours)
     return Value(static_cast<double>(this_object->hours()));
 }
 
+JS_DEFINE_NATIVE_FUNCTION(DatePrototype::set_hours)
+{
+    auto* this_object = typed_this(vm, global_object);
+    if (!this_object)
+        return {};
+
+    auto new_hours = vm.argument(0).to_i32(global_object);
+    if (vm.exception())
+        return {};
+
+    auto& datetime = this_object->datetime();
+
+    i32 new_minutes;
+    if (vm.argument_count() >= 2) {
+        new_minutes = vm.argument(1).to_i32(global_object);
+        if (vm.exception())
+            return {};
+    } else {
+        new_minutes = datetime.minute();
+    }
+
+    i32 new_seconds;
+    if (vm.argument_count() >= 3) {
+        new_seconds = vm.argument(2).to_i32(global_object);
+        if (vm.exception())
+            return {};
+    } else {
+        new_seconds = datetime.second();
+    }
+
+    if (vm.argument_count() >= 4) {
+        auto new_milliseconds = vm.argument(3).to_i32(global_object);
+        if (vm.exception())
+            return {};
+        new_seconds += new_milliseconds / 1000;
+        this_object->set_milliseconds(new_milliseconds % 1000);
+    }
+
+    datetime.set_time(datetime.year(), datetime.month(), datetime.day(), new_hours, new_minutes, new_seconds);
+    return Value { this_object->time() };
```

Comment:
Please use `Value()` for consistency with 99.99% of LibJS (might as well fix the 0.01% of `Value {}` @awesomekling introduced recently :stuck_out_tongue:)